zebra/zebra_rnh.c

```diff
@@ -1426,10 +1426,19 @@ void show_nexthop_json_helper(json_object *json_nexthop,
 		json_object_object_add(json_nexthop, "seg6local",
 				       json_seg6local);
 
-		json_seg6 = json_object_new_object();
-		json_object_string_addf(json_seg6, "segs", "%pI6",
-					&nexthop->nh_srv6->seg6_segs);
-		json_object_object_add(json_nexthop, "seg6", json_seg6);
+		json_segs = json_object_new_array();
```

Comment:
and breaking existing json consumers right? <sigh>

Reply:
done, when seg_num == 1 we use standard procedure, else we apply seg list. In this wawy we keep backwards compatibility with json consumers.